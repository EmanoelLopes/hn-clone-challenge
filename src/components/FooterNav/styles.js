import styled from 'styled-components';

export const Navigation = styled.nav`
  a {
    color: #e74c3c;
    display: inline-block;
    font-size: 14px;
    margin: 0 .5rem;

    &:hover {
      text-decoration: underline;
    }
  }
`;

export const Copyrights = styled.p`
  color: #999;
  font-size: 14px;
  margin: .5rem 0;
  text-align: center;
`;