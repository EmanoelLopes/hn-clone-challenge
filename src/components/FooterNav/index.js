import React from 'react';
import { Navigation, Copyrights } from './styles';

function FooterNav () {
  return (
    <div className="footer-nav">
      <Navigation>
        <a href="/" title="About">About</a>
        <a href="/" title="Contact">Contact</a>
        <a href="/" title="Sign In">Sign In</a>
      </Navigation>
      <Copyrights>&copy; 2014 - The UX Library</Copyrights>
    </div>
  );
};

export default FooterNav;