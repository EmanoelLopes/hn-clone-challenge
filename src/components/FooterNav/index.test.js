import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import FooterNav from '../FooterNav';

configure({ adapter: new Adapter() });

describe('<FooterNav />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<FooterNav />);
    expect(wrapper).toMatchSnapshot();
  });
});

