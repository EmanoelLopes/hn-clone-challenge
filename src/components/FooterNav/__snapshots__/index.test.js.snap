// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FooterNav /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FooterNav />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <styled.nav>
          <a
            href="/"
            title="About"
          >
            About
          </a>
          <a
            href="/"
            title="Contact"
          >
            Contact
          </a>
          <a
            href="/"
            title="Sign In"
          >
            Sign In
          </a>
        </styled.nav>,
        <styled.p>
          © 2014 - The UX Library
        </styled.p>,
      ],
      "className": "footer-nav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <a
              href="/"
              title="About"
            >
              About
            </a>,
            <a
              href="/"
              title="Contact"
            >
              Contact
            </a>,
            <a
              href="/"
              title="Sign In"
            >
              Sign In
            </a>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "About",
              "href": "/",
              "title": "About",
            },
            "ref": null,
            "rendered": "About",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Contact",
              "href": "/",
              "title": "Contact",
            },
            "ref": null,
            "rendered": "Contact",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign In",
              "href": "/",
              "title": "Sign In",
            },
            "ref": null,
            "rendered": "Sign In",
            "type": "a",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "© 2014 - The UX Library",
        },
        "ref": null,
        "rendered": "© 2014 - The UX Library",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <styled.nav>
            <a
              href="/"
              title="About"
            >
              About
            </a>
            <a
              href="/"
              title="Contact"
            >
              Contact
            </a>
            <a
              href="/"
              title="Sign In"
            >
              Sign In
            </a>
          </styled.nav>,
          <styled.p>
            © 2014 - The UX Library
          </styled.p>,
        ],
        "className": "footer-nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <a
                href="/"
                title="About"
              >
                About
              </a>,
              <a
                href="/"
                title="Contact"
              >
                Contact
              </a>,
              <a
                href="/"
                title="Sign In"
              >
                Sign In
              </a>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "About",
                "href": "/",
                "title": "About",
              },
              "ref": null,
              "rendered": "About",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Contact",
                "href": "/",
                "title": "Contact",
              },
              "ref": null,
              "rendered": "Contact",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign In",
                "href": "/",
                "title": "Sign In",
              },
              "ref": null,
              "rendered": "Sign In",
              "type": "a",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "© 2014 - The UX Library",
          },
          "ref": null,
          "rendered": "© 2014 - The UX Library",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
